text/0000-otlp-proto-version-identifier.md

```diff
@@ -0,0 +1,31 @@
+# Include OTLP proto version identifier in requests
+
+Adds OTLP request identifer to descrive the OTLP version used to create the binary or JSON payload.
+
+## Motivation
+
+It is useful to know the version of the OTLP protobuf definition used to create the binary or JSON.
+
+This would enable routing to different services by inspecting the header value. This is especially useful for organisations that wish to support multiple versions where the OTLP proto definitions have changed. A currently used alternative is to append the version to the path, but this is clumsy and requires additional configuration when setting up the exporter.
+
+## Explanation
+
+Currently it is not possible to know the version of the OTLP proto definition before receving it. This is challening because there have been changes to the OTLP proto specification that are breaking.
```

Comment:
As an example, *even between stable OTLP versions*, if a 0.16 OTLP handler receives data encoded for v0.9, it will receive data that is no longer recognized (IntSum, IntGauge, IntHistogram, and Labels fields have all been removed by now). The upgrade is dangerous because the payload has no metadata to indicate its version, so once I remove the v0.9 handler I'll have no way to tell data is being dropped.